R/ggscatter.R

```diff
@@ -9,25 +9,38 @@ library(ggplot2)
 #' @param df Dataframe to plot
 #' @param xval x-var Column name used as the x-axis variable
 #' @param yval y-var Column name used as the y-axis variable
+#' @param x_transform Determines whether the x-axis undergoes a natural log transformation
+#' @param y_transform Determines whether the y-axis undergoes a natural log transformation
 #'
 #' @return ggplot Object
 #' @export
 #'
 #' @examples
-#' scatter_express(hiris, "Sepal.Width, Petal.Length)
-scatter_express <- function(df, xval = NA, yval = NA){
+#' scatter_express(iris, "Sepal.Width, Petal.Length)
+scatter_express <- function(df, xval = NA, yval = NA, x_transform = FALSE, y_transform = FALSE){
 
   corr_df <- dplyr::select(df, {{xval}}, {{yval}})
+
+  if (is.numeric(df[[1]]) == FALSE) stop("Your x-variable must be numeric")
+  if (is.numeric(df[[2]]) == FALSE) stop("Your y-variable must be numeric")
+
   corr_val <- round(stats::cor(df[[1]], df[[2]]), 2)
 
 
-  scatter <- ggplot2::ggplot(df, ggplot2::aes(x = {{xval}}, y = {{yval}})) + ggplot2::geom_point()
+  scatter <- ggplot2::ggplot(df, ggplot2::aes(x = {{xval}}, y = {{yval}})) +
+    ggplot2::geom_point(color = "blue")
 
-  scatter <- scatter + ggplot2::labs(title = paste(rlang::get_expr(scatter$mapping$x), " vs ", rlang::get_expr(scatter$mapping$y), "(Correlation: ",  corr_val, ")"))
-  scatter
-}
+  if (x_transform == TRUE){
+    scatter <- scatter + ggplot2::scale_x_continuous(trans= "log2")
 
+  }
 
-df <- iris
+  if (y_transform == TRUE){
+    scatter <- scatter + ggplot2::scale_y_continuous(trans = "log2")
+  }
+
+  scatter <- scatter + ggplot2::labs(title = paste(rlang::get_expr(scatter$mapping$x), " vs ", rlang::get_expr(scatter$mapping$y), "(Pearson Correlation: ",  corr_val, ")"))
+  scatter
```

Comment:
-  could add more brief comments to improve readability and future de-bugging. 
- May want to allow the user to customize their axis labels and plot title.